example/src/screens/EmbeddedPaymentElementScreen.tsx

```diff
@@ -0,0 +1,270 @@
+            light: {
+              background: actionPink,
+              text: brandWhite, // White text on pink
+              border: '00000000', // No border color needed
+            },
+            dark: {
+              background: actionPink,
+              text: brandWhite, // White text on pink
+              border: '#00000000', // No border color needed
+            },
+          },
+        },
+        embeddedPaymentElement: {
+          row: {
+            style: RowStyle.FloatingButton,
+            floating: {
+              spacing: 12,
+            },
+          },
+        },
+      };
+
+      // 4. Element UI config
+      const uiConfig: EmbeddedPaymentElementConfiguration = {
+        merchantDisplayName: 'Example Inc.',
+        returnURL: 'stripe-example://stripe-redirect',
+        customerId: customer,
+        defaultBillingDetails: billingDetails,
+        defaultShippingDetails: shippingDetails,
+        allowsDelayedPaymentMethods: true,
+        formSheetAction: {
+          type: 'confirm',
+          onFormSheetConfirmComplete: (
+            result: EmbeddedPaymentElementResult
+          ) => {
+            if (result.status === 'completed')
+              Alert.alert('Success', 'Payment confirmed');
+            else if (result.status === 'failed')
+              Alert.alert('Error', `Failed: ${result.error.message}`);
+            else Alert.alert('Cancelled');
+          },
+        },
+        rowSelectionBehavior: {
+          type: 'immediateAction',
+          didSelectPaymentOption: () => console.log('Option selected'),
+        },
+        appearance,
+      };
+
+      // 5. Intent config
+      const newIntentConfig: IntentConfiguration = {
+        confirmHandler: async (
+          paymentMethod: PaymentMethod.Result,
+          _save,
+          callback: (res: IntentCreationCallbackParams) => void
+        ) => {
+          const resp = await fetch(
+            `${API_URL}/payment-intent-for-payment-sheet`,
+            {
+              method: 'POST',
+              headers: { 'Content-Type': 'application/json' },
+              body: JSON.stringify({
+                paymentMethodId: paymentMethod.id,
+                customerId: customer,
+              }),
+            }
+          );
+          const { clientSecret, error } = await resp.json();
+          if (error)
+            callback({
+              error: {
+                code: 'Failed',
+                message: error.raw.message,
+                localizedMessage: error.raw.message,
+              },
+            });
+          else callback({ clientSecret });
+        },
+        mode: { amount: 6099, currencyCode: 'USD' },
+        paymentMethodTypes: ['card', 'klarna'],
+      };
+
+      setElementConfig(uiConfig);
+      setIntentConfig(newIntentConfig);
+    },
+    [customerKeyType]
+  );
+
+  // Hook into Stripe element
+  const { view, paymentOption, confirm, clear } = useEmbeddedPaymentElement(
+    intentConfig!,
```

Comment:
Can we allow passing null / undefined here? I think generally non-null assertion is seen as bad.